src/aws/require-aws-bare-bones.ts

```diff
@@ -6,31 +6,48 @@ import getDocumentationUrl from '../get-documentation-url.ts';
 export const ruleId = 'require-aws-bare-bones';
 export const MESSAGE_ID_AGGREGATED_CLIENT = 'noAggregatedClient';
 
-const BARE_BONES_SUFFIXES = /(?:Client|Command|Exception|Input|Output)$/u;
+const BARE_BONES_SUFFIXES = new Set(['Client', 'Command', 'Exception', 'Input', 'Output']);
+const AWS_LIB_AGGREGATED_SUFFIXES = new Set(['Document']);
+const AWS_SDK_CLIENT = '@aws-sdk/client-';
+const AWS_SDK_LIB = '@aws-sdk/lib-';
 
-const createRule = ESLintUtils.RuleCreator((name) => getDocumentationUrl(name));
+const createRule = ESLintUtils.RuleCreator(getDocumentationUrl);
 
-function isAwsSdkClientModule(importDeclaration: TSESTree.ImportDeclaration): boolean {
-  return (
-    typeof importDeclaration.source.value === 'string' && importDeclaration.source.value.startsWith('@aws-sdk/client-')
-  );
-}
+const isAwsSdkClientModule = ({ source }: TSESTree.ImportDeclaration): boolean =>
+  typeof source.value === 'string' && (source.value.startsWith(AWS_SDK_CLIENT) || source.value.startsWith(AWS_SDK_LIB));
 
-function isAggregatedClient(name: string): boolean {
-  return !BARE_BONES_SUFFIXES.test(name);
-}
+const kebabToPascal = (str: string): string =>
+  str
+    .split('-')
+    .map((item) => item.charAt(0).toUpperCase() + item.slice(1))
+    .join('');
+
+const endsWithAnySuffix = (name: string, suffixes: Set<string>): boolean =>
+  Array.from(suffixes).some((suffix) => name.endsWith(suffix));
+
+const isAggregatedClient = (name: string, importSource: string): boolean => {
+  if (importSource.startsWith(AWS_SDK_CLIENT)) {
+    return !endsWithAnySuffix(name, BARE_BONES_SUFFIXES);
+  }
+  if (importSource.startsWith(AWS_SDK_LIB)) {
+    const pkg = importSource.replace(AWS_SDK_LIB, '');
```

Comment:
should BARE_BONES_SUFFIXES be checked for lib-* stuff as well?
btw the naming pattern matching for lib stuff feels a bit tricky, but hopefully it's good enough to cover what we normally use.